add features.md and fuel-core lib doc. (#480)
FuelLabs/fuels-rs: docs/src/providers/short-lived.md

```rust,ignore
let wallet = launch_provider_and_get_wallet().await;
```

# Features

> ### Fuel-core lib

Reviewer: ```suggestion
## Fuel-core lib
```

> The `fuel-core-lib` is a feature defined in `fuels` library, and it allows us to run a fuel-core node without installing the fuel-core binary on the local machine. This entails downloading all the dependencies needed to run the fuel-core node.

Reviewer: Clarity improvements:

```suggestion
The `fuel-core-lib` is a feature defined in the `fuels` library, allowing us to run a `fuel-core` node without installing the `fuel-core` binary on the local machine. Using the `fuel-core-lib` feature flag entails downloading all the dependencies needed to run the fuel-core node.
```
